pkg/syncer/syncer.go

```diff
@@ -73,15 +73,58 @@ func (sc *SyncerConfig) ID() string {
 func StartSyncer(ctx context.Context, cfg *SyncerConfig, numSyncerThreads int, importPollInterval time.Duration) error {
 	klog.Infof("Starting syncer for logical-cluster: %s, workload-cluster: %s", cfg.KCPClusterName, cfg.WorkloadClusterName)
 
+	kcpClusterClient, err := kcpclient.NewClusterForConfig(rest.AddUserAgent(rest.CopyConfig(cfg.UpstreamConfig), "kcp#syncer/v0.0.0"))
+	if err != nil {
+		return err
+	}
+
+	// For now we retrieve the syncerVirtualWorkpaceURL at start, since we temporarily stick to a single URL (sharding not supported).
+	// But the complete implementation should setup a WorkloadCluster informer, create spec and status syncer for every URLs found in the
+	// Status.SyncerVirtualWorkspaceURLs slice, and update them each time this list changes.
+	var syncerVirtualWorkspaceURL string
+	err = wait.PollImmediateInfinite(gvrQueryInterval, func() (bool, error) {
+		klog.Infof("Attempting to retrieve the Syncer virtual workspace URL from WorkloadCluster %s|%s", cfg.KCPClusterName, cfg.WorkloadClusterName)
+
+		workloadCluster, err := kcpClusterClient.Cluster(cfg.KCPClusterName).WorkloadV1alpha1().WorkloadClusters().Get(ctx, cfg.WorkloadClusterName, metav1.GetOptions{})
+		if err != nil {
+			return false, err
```

Comment:
> If the WorkloadCluster doesn't exist, err will be NotFound, and this will exit the polling loop with an error... and then on line 104, it says "Should never happen". Would recommend either changing this logic to be accepting of NotFound errors, or adjust the comment below.

The "Should never happen" comment was just a wrong copy-paste. Let me remove it and keep the same logic (error when WorkloadCluster is not found), since this is already applied later on, in line [177](https://github.com/kcp-dev/kcp/blob/fd2d0affb194438440282df4f340b922e0fd57b4/pkg/syncer/syncer.go#L177)

> Also, if this is polling forever, the user has no idea what is going on unless they look in the syncer pod logs. Would recommend setting a condition on the WorkloadCluster before entering this poll loop indicating current status. Could do as a TODO (follow-up) or in this PR

I'd rather add a TODO and leave that for a next PR, since it seems to be the same need as what is in line [148](https://github.com/kcp-dev/kcp/blob/fd2d0affb194438440282df4f340b922e0fd57b4/pkg/syncer/syncer.go#L148).

In addition to that, I assume that when we start managing the list of URLs, we'll have to refactor this deeply.

Reply:
Dummy comment removed